### PR TITLE
Evolutions

### LINKED ISSUE #34: Automatic evolutions
There should be a flag called --evolve-pokemon which enables automatic evolutions of Pokemon. 
When the flag is active, whenever the player acquires enough candy to evolve a pokemon, it should evolve the highest CP pokemon. This can also be paired with --initial-evolve which iterates over all of the of the current pokemon and evolves any that can be evolved. The default for these two arguments should be False.

### PR DESCRIPTION
**Pull request by epicblood**
### Short Description:

Bot will evolve pokemon when it has enough candies for it.
It runs after_catch_pokemon and after_transfer_pokemon.
### Changes:
- Added list of candies to bot
- Added method to update/add candies
- Moved `_initial_transfer_get_groups` to bot as `get_pokemon`
- Added Evolver plugin
- Added 'after_transfer_pokemon' event
- Added evolve_pokemon config

Closes #34 

@OpenPoGo/maintainers

### CODE REVIEW

**Comment by MaxLeiter** [comment]
Awesome work so far, but I'd really like to see some config options for this. Is it feasible to have the user (in the config file) specify which pokemon to evolve? (or, alternately, which Pokemon not to evolve?) 

Basically, I'd like it so if `-ev` is specified on the commandline, it uses whatever rules are in the json file, and if no rules are present, it evolves whenever possible.

**Comment by epicblood** [comment]
I can work on that, about to get on a plane so wont be till tomorrow at least.

**Comment by MaxLeiter** [comment]
No rush, thanks for the work so far

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -205,6 +231,29 @@ def update_player_and_inventory(self):
         self.api_wrapper.get_player().get_inventory()
         return self.api_wrapper.call()
 
+    def update_candies(self, candies):
+        for key in candies:
+            self.candies[key] = candies[key]
+        self._candies_use_name()
+
+    def add_candies(self, name=None, pokemon_candies=None):
```
Comment (andythorne) [inline pokemongo_bot/__init__.py:186]: Is this ever called?

Comment (epicblood) [inline pokemongo_bot/__init__.py:186]: It was, need to refactor a bit still after the api wrapper was merged.

**Pull request closed without merging**